source/patterns/@aws-solutions-constructs/aws-lambda-kinesisstreams/test/lambda-kinesisstream.test.ts

```diff
@@ -0,0 +1,495 @@
+  const kinesisStream = template.findResources('AWS::Kinesis::Stream');
+  const [ kinesisStreamId ] = Object.keys(kinesisStream);
+
+  expect(stack).toHaveResourceLike('AWS::IAM::Policy', {
+    PolicyDocument: {
+      Statement: [
+        {
+          Action: [
+            'xray:PutTraceSegments',
+            'xray:PutTelemetryRecords'
+          ],
+          Effect: 'Allow',
+          Resource: '*'
+        },
+        {
+          Action: [
+            'kinesis:ListShards',
+            'kinesis:PutRecord',
+            'kinesis:PutRecords'
+          ],
+          Effect: 'Allow',
+          Resource: {
+            'Fn::GetAtt': [
+              kinesisStreamId,
+              'Arn'
+            ]
+          }
+        }
+      ]
+    }
+  });
+});
+
+test('When a Customer-manged CMK is used on an existing stream, construct grants the Lambda Function permission to use the encryption key so it can publish messages to it', () => {
+  const stack = new Stack();
+
+  const encryptionKey = new kms.Key(stack, 'Key', {
+    description: 'my-key-description'
+  });
+
+  const existingStreamObj = new kinesis.Stream(stack, 'test-stream', {
+    streamName: 'my-stream',
+    encryptionKey
+  });
+
+  new LambdaToKinesisStreams(stack, 'test-lambda-kinesisstreams', {
+    lambdaFunctionProps: {
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: 'index.handler'
+    },
+    existingStreamObj
+  });
+
+  const template = Template.fromStack(stack);
+  const resource = template.findResources('AWS::KMS::Key', {
+    Properties: {
+      Description: Match.exact('my-key-description')
+    }
+  });
+  const [ kmsKey ] = Object.keys(resource);
+
+  expect(stack).toHaveResourceLike('AWS::IAM::Policy', {
+    PolicyDocument: {
+      Statement: [
+        {
+          Action: [
+            'xray:PutTraceSegments',
+            'xray:PutTelemetryRecords'
+          ],
+          Effect: 'Allow',
+          Resource: '*'
+        },
+        {
+          Action: [
+            'kinesis:ListShards',
+            'kinesis:PutRecord',
+            'kinesis:PutRecords'
+          ],
+          Effect: 'Allow',
+          Resource: {
+            'Fn::GetAtt': [
+              'teststream04374A09',
+              'Arn'
+            ]
+          }
+        },
+        {
+          Action: [
+            'kms:Encrypt',
+            'kms:ReEncrypt*',
+            'kms:GenerateDataKey*'
+          ],
+          Effect: 'Allow',
+          Resource: {
+            'Fn::GetAtt': [
+              kmsKey,
+              'Arn'
+            ]
+          }
+        },
+        {
+          Action: [
+            'kms:Decrypt',
+            'kms:GenerateDataKey*'
+          ],
+          Effect: 'Allow',
+          Resource: {
+            'Fn::GetAtt': [
+              kmsKey,
+              'Arn'
+            ]
+          }
+        }
+      ]
+    }
+  });
+});
+
```

Comment:
Let's add a test for one or both unencrypted stream situations (existing stream that's not encrypted or streamProps that turn off encryption)

Reply:
will do!